### PR TITLE
new p2p

### PR DESCRIPTION
**Pull request by lochjin**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,28 +4,48 @@ go 1.12
 
 require (
 	github.com/Qitmeer/crypto v0.0.0-20200516043559-dd457edff06c
+	github.com/aristanetworks/goarista v0.0.0-20200812190859-4cb0e71f3c0e
+	github.com/btcsuite/btcd v0.20.1-beta
```
Comment (dindinw) [inline go.mod:8]: should not depend on `btcd`

File: go.mod [hunk 2]
```diff
@@ -4,28 +4,48 @@ go 1.12
...
 	github.com/cloudflare/roughtime v0.0.0-20200829152512-a9bb6267a4f5
 	github.com/davecgh/go-spew v1.1.1
 	github.com/dchest/blake256 v1.0.0
 	github.com/deckarep/golang-set v1.7.1
+	github.com/dgraph-io/ristretto v0.0.2
+	github.com/ferranbt/fastssz v0.0.0-20200514094935-99fccaf93472
 	github.com/go-stack/stack v1.8.0
+	github.com/gogo/protobuf v1.3.1
 	github.com/golang-collections/collections v0.0.0-20130729185459-604e922904d3
+	github.com/golang/protobuf v1.4.0
```
Comment (dindinw) [inline go.mod:18]: depends on two `protobuf`

**Review by dindinw — CHANGES_REQUESTED** [review]
Please double check the `go.mod`

File: go.mod (hunk 2 above)
Comment (lochjin) [reply go.mod:17]: A combination of the two is needed

File: go.mod (hunk 1 above)
Comment (lochjin) [reply go.mod]: https://github.com/Qitmeer/qitmeer/pull/382/commits/247260ab15756863809d6622ac5c1147b3cf2f77

**Comment by dindinw** [comment]
Please fix the security warning
https://github.com/Qitmeer/qitmeer/security/code-scanning/7?query=ref%3Arefs%2Fpull%2F382%2Fhead

**Comment by lochjin** [comment]
> Please fix the security warning
> https://github.com/Qitmeer/qitmeer/security/code-scanning/7?query=ref%3Arefs%2Fpull%2F382%2Fhead

https://github.com/Qitmeer/qitmeer/pull/382/commits/74425266d6669e29bd89811a5eaf206d7099f7b5

**Review by dindinw — APPROVED** [review]

**Pull request merged**